packages/core/src/internal/execution/types/execution-strategy.ts

```diff
@@ -216,7 +218,15 @@ export interface ExecutionStrategy {
    */
   name: string;
 
-  init: () => Promise<void>;
+  /**
+   * The configuration options for the strategy.
+   */
+  config: Record<PropertyKey, string | number> | Record<PropertyKey, never>;
```

Comment:
This union lets us cast from the reduced strategy types to the fuller strategy types.